test/testframe.c

```diff
@@ -147,28 +176,66 @@ AddTest(const char *TestName, void (*TestFunc)(void *), void (*TestSetupFunc)(vo
 
         memcpy(new_test_data, TestData, TestDataSize);
     }
-    TestArray[TestCount].TestParameters = new_test_data;
+    TestArray[TestCount].TestParameters.TestParams     = new_test_data;
+    TestArray[TestCount].TestParameters.TestParamsSize = TestDataSize;
 
     TestArray[TestCount].TestFunc        = TestFunc;
     TestArray[TestCount].TestSetupFunc   = TestSetupFunc;
     TestArray[TestCount].TestCleanupFunc = TestCleanupFunc;
+    TestArray[TestCount].HeaderFunc      = NULL;
 
-    TestArray[TestCount].TestFrameworkFlags = TestFrameworkFlags;
+    TestArray[TestCount].TestFlags = TestFlags;
 
-    H5_ATOMIC_STORE(TestArray[TestCount].TestNumErrors, -1);
+    H5_ATOMIC_STORE(TestArray[TestCount].TestNumErrors, 0);
 
     TestCount++;
 
     return SUCCEED;
 }
 
+/*
+ * Add a header function to a test that will run before the test
+ */
+herr_t
+AddTestHeaderFunc(const char *TestName, void (*HeaderFunc)(TestParams_t *))
```

Comment:
This was added specifically for the API tests to be able to properly print a banner between groups of tests when using the testing framework.